### PR TITLE
Added unique_environment_block to resource.h

### PR DESCRIPTION
**Pull request by m417z**
Fixes https://github.com/microsoft/wil/issues/189

### CODE REVIEW

File: include/wil/resource.h [hunk 1]
```diff
@@ -5361,6 +5361,11 @@ namespace wil
     typedef unique_any<HTHEME, decltype(&::CloseThemeData), ::CloseThemeData> unique_htheme;
 #endif // __WIL_INC_UXTHEME
 
+#if defined(_INC_USERENV) && !defined(__WIL_INC_USERENV) && !defined(WIL_KERNEL_MODE)
```
Comment (sylveon) [inline include/wil/resource.h:5364]: I think you need a check for `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)` as well.

Comment (m417z) [reply include/wil/resource.h]: Done, thanks.

**Review by sylveon — APPROVED** [review]
LGTM, I'd add the shared_ and weak_ variants as well but they may not be very useful for environment blocks.

File: include/wil/resource.h (hunk 1 above)
Comment (jonwis) [reply include/wil/resource.h]: I look forward to the day when we can get rid of that entirely. :|

**Review by jonwis — APPROVED** [review]

**Review by dunhor — APPROVED** [review]

**Pull request merged**